### DISCUSSION TITLE
[rewrite for single-file] add a new flag for enable_rewrite_for_single_file vs. single-file mode overwrites the existing resolver.go but build time breaking experience?

### DISCUSSION THREAD

**Discussion by hailyngx** [99designs/gqlgen#3205] [Q&A]
## What happened?
I wanted to use the `rewrite` for single-file mode and made direct changes to `generateSingleFile`. I made changes in very similar fashion to `generatePerSchema` so it behaved similarly in terms of rewriting the file. However, when I run `go generate` at the root of gqlgen folder. One of the folder insider `_examples` broke, specifically `type-system-extension`. 

## What did you expect?
I understand that the current `generateSingleFile` does not support rewriting in single-file mode. Still, I want to integrate `resolvergen` to a build system that cannot work with `generatePerSchema` but maintaining the rewrite function is important for large schema changes.

## Potential Solution
Can we add a new flag `enable_rewrite_for_single_file` and build on top of that? By doing so, we will not break any build from the `_examples` while allowing for some flexibility in using single-file. The default for this flag is false but we can use it by adding to our config file:
```
resolver:
  filename: resolver/resolver.go
  enable_rewrite_for_single_file: true
```

### `codegen/config/resolver.go`
* Add a new boolean field `EnableRewriteForSingleFile` inside `ResolverConfig` struct.

### `plugin/resolvergen/resolver.go`
* Add a new `generateSingleFileEnableRewrite` function with similar rewrite behaviors to `generatePerSchema`
* Inside `GenerateCode`
```
func (m *Plugin) GenerateCode(data *codegen.Data) error {
 	switch data.Config.Resolver.Layout {
 	case config.LayoutSingleFile:
		if data.Config.Resolver.EnableRewriteForSingleFile { // add a conditional statement to use rewrite for single-file
			return m.generateSingleFileEnableRewrite(data)
	}
 		return m.generateSingleFile(data)
 	case config.LayoutFollowSchema:
 		return m.generatePerSchema(data)
```
* Change file notice: `// THIS CODE WILL BE UPDATED WITH SCHEMA CHANGES. PREVIOUS IMPLEMENTATION FOR SCHEMA CHANGES WILL BE KEPT IN THE COMMENT SECTION. IMPLEMENTATION FOR UNCHANGED SCHEMA WILL BE KEPT.`
* Added unit test and `testdata` to cover the new function `GenerateSingleFileEnableRewrite`.

Would appreciate any input! 💡 Thank you for your time.

**Comment by StevenACoffman** [comment]
See https://github.com/99designs/gqlgen/pull/3243
